### ISSUE TITLE
Correct Normalization Values for CIFAR-10

### ISSUE THREAD

**Issue by dlmacedo** [kuangliu/pytorch-cifar#19]
Correct values for CIFAR-10: mean=[0.507, 0.486, 0.440], std=[0.267, 0.256, 0.276]

**Comment by tyunist** [comment]
Great! It works much better than the previous one. 
Could you explain more about how did you compute those values?
Thanks

**Comment by Armour** [comment]
@tyunist  Hi, [here](https://github.com/Armour/pytorch-nn-practice/blob/master/utils/meanstd.py) is my code for how to compute those value, hope it helps :)

**Comment by tyunist** [comment]
Great! Thanks, Armour.  
Based on your code, I found out a bug in my code causing worse performance after standardization.

**Comment by Armour** [comment]
Nice, I'm glad I could help :)

**Comment by netaz** [comment]
Hi guys,
Any idea why the pytorch CIFAR10 [example](http://pytorch.org/tutorials/beginner/blitz/cifar10_tutorial.html) uses:
transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5)?

And while I'm asking, why do you do:
transforms.RandomCrop(32, padding=4)?

I.e. aren't the images already 32x32?  And why pad the images?
Thanks!
Neta

**Comment by luuuyi** [comment]
@netaz I think after set padding=4 the image become 40 x 40, random crop operator can get more result. If we do not set padding=4, random crop only get origin image.

**Comment by nzmora** [comment]
@luuuyi yes, I agree with you.  This occurred to me some time after I posted the question, but I didn't think to update here.  Thanks for answer!

**Comment by gobbedy** [comment]
the behaviour of ```RandomCrop(32, padding=4)``` still wasn't clear to me after reading the documentation and reading this issue, so I did a little digging.

It order, it does
1. pads the original image on all sides with 4 pixels, resulting in a 40x40 image (since the original image is 32x32 in cifar10)
2. takes a random 32x32 crop of this 40x40 image

The result is weird black lines around a chopped off version of the original image

eg, original:
![image](https://user-images.githubusercontent.com/26829704/59960434-4b3ffa00-9496-11e9-9a7c-e3558d55a050.png)

after ```RandomCrop(32, padding=4)``` transformation:
![image](https://user-images.githubusercontent.com/26829704/59960400-da98dd80-9495-11e9-91e4-0ed02406e46e.png)

**Comment by parkie0517** [comment]
Thank you @gobbedy. this is very helpful!
